viewing_party/party.py

```diff
@@ -0,0 +1,259 @@
+        
+
+#Wave_Three Part 1 
+
+def get_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    movie_list = []
+    user_movie_set = set()
+    friend_movie_set = set()
+    for movie in user_watched: 
+        user_movie_set.add(movie["title"])
+        print(user_movie_set)
+    for friend in friends_watched:
+        for friends_watched_list in friend.values():
+                # print(friends_watched_list)
+            for movie_title in friends_watched_list:
+                    # print(movie_title)
+                    friend_movie_set.add(movie_title["title"])
+
+    potential_friends_watched_set = (user_movie_set) - (friend_movie_set) 
+    final_movie_list = []
+    for potential in potential_friends_watched_set:
+        final_movie_list.append({"title":potential})
+    return final_movie_list
+
+
+
+    
+    
+
+#Wave 3 Part 2 
+
+def get_friends_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    movie_list = []
+    user_movie_set = set()
+    friend_movie_set = set()
+    for movie in user_watched: 
+        user_movie_set.add(movie["title"])
+        print(user_movie_set)
+    for friend in friends_watched:
+        for friends_watched_list in friend.values():
+                # print(friends_watched_list)
+            for movie_title in friends_watched_list:
+                    # print(movie_title)
+                    friend_movie_set.add(movie_title["title"])
+    
+    friends_unique_movies = []
+    friends_unique_movie_set = (friend_movie_set) - (user_movie_set)
+
+    for movie in friends_unique_movie_set:
+        friends_unique_movies.append({"title":movie})
+    return friends_unique_movies
+
+
+            
+        
+
+
+
+#Wave 4 Part1
+
+
+
+
+def get_available_recs(user_data):
+    users_subscription = user_data["subscriptions"]
+    friends_unique_movies = get_friends_unique_watched(user_data) 
+    users_friends = user_data["friends"]
+    users_friends_watched_movie = user_data["friends"]
+    
+    
+    friends_movie_list = []
+    for watched_movies in users_friends:
+    
+        
+        for movies in watched_movies["watched"]:
+            friends_movie_list.append(movies)
+
+    movie_subscription = []
+    for movies_watched in friends_movie_list:
+        if movies_watched["host"] in users_subscription and movies_watched not in movie_subscription:
+            movie_subscription.append(movies_watched)
+    print(movie_subscription)
+    
+
+    user_title =[]
+    for watched_movie in user_data["watched"]:
+        user_title.append(watched_movie["title"])
+
+
+    recommendations =[]
+
+    if not user_data["watched"]:
+        return movie_subscription
+    for movie in movie_subscription:
+        if movie["title"] not in user_title:
+            recommendations.append(movie)
+    return recommendations
```

Comment:
This is a good implementation but I wanted to offer a solution that uses less lines of code
```suggestion
def get_available_recs(user_data):
    recommended_movies = []
    friends_unique_watched = get_friends_unique_watched(user_data)

    for movie in friends_unique_watched:
        if movie["host"] in user_data["subscriptions"]:
            recommended_movies.append(movie)   
    return recommended_movies
```